src/tainting/Dataflow_tainting.ml

```diff
@@ -1271,7 +1272,15 @@ let lval_of_sig_arg fun_exp fparams args_exps (sig_arg : T.arg) =
          e = Fetch { base = Var obj; rev_offset = [ { o = Dot _method; _ } ] };
          _;
         } ->
+            (* We're calling `obj.method`, so `this.x` is actually `obj.x` *)
             Some ({ base = Var obj; rev_offset = List.rev os }, obj)
+        | { e = Fetch { base = Var method_; rev_offset = [] }; _ } ->
+            (* We're calling a `method` on the same instace of the caller,
+             * and `this.x` it's just `this.x` *)
+            let this =
+              VarSpecial (This, Tok.fake_tok (snd method_.ident) "this")
+            in
+            Some ({ base = this; rev_offset = List.rev os }, method_)
```

Comment:
Why is `method_` used as the `name` portion of the return type? Is it because it has a real token?

Also I'm wondering if we need `name` or are we just interested in the token. I've tried finding the callers of this function. One caller disregards this returned name. Another uses the token part of the name.

In general, I think I probably would benefit from some documentation for the `name` part of the return type. The function name only mentions lval, so I was expecting the return type to be just an lval.

Reply:
Fair point, yeah it's only used for the purpose of having a taint trace. I'll try to improve this!